src/zenml/integrations/wandb/experiment_trackers/wandb_experiment_tracker.py

```diff
@@ -131,25 +127,50 @@ def cleanup_step_run(self, info: "StepRunInfo", step_failed: bool) -> None:
 
     def _initialize_wandb(
         self,
+        info: "StepRunInfo",
         run_name: str,
         tags: List[str],
-        settings: Union["Settings", Dict[str, Any], None] = None,
     ) -> None:
         """Initializes a wandb run.
 
         Args:
+            info: Step run information.
             run_name: Name of the wandb run to create.
             tags: Tags to attach to the wandb run.
-            settings: Additional settings for the wandb run.
         """
         logger.info(
             f"Initializing wandb with entity {self.config.entity}, project "
             f"name: {self.config.project_name}, run_name: {run_name}."
         )
+        settings = cast(
+            WandbExperimentTrackerSettings, self.get_settings(info)
+        )
         wandb.init(
             entity=self.config.entity,
             project=self.config.project_name,
             name=run_name,
             tags=tags,
-            settings=settings,
+            settings=settings.settings,
         )
+
+        if settings and settings.enable_weave:
+            import weave
+
+            logger.info("Initializing weave")
```

Comment:
Maybe this log could be in the `if self.config.project_name` to not clutter the logs?